app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt

```diff
@@ -50,47 +60,116 @@ class WalletFragment : BaseFragment(), OnLoadAccount {
         endPollingAccount()
     }
 
+    //region User Interface
+
     private fun setupUI() {
+        bindAdapter()
         loadBalance()
     }
 
-    private fun loadBalance() {
+    private fun bindAdapter() {
+        recyclerViewArrayList = WalletHeterogenousArray(TotalBalance(loadBalance()),
+                AvailableBalance(loadBalance()), Pair("Activity", "Amount"), effectsList)
+
+        adapter = WalletRecyclerViewAdapter(activity!!, recyclerViewArrayList!!.array)
+        adapter!!.setOnAssetDropdownListener(object : WalletRecyclerViewAdapter.OnAssetDropdownListener {
+            override fun onAssetDropdownClicked(view: View, position: Int) {
+                startActivity(Intent(activity, WalletsActivity::class.java))
+                activity?.overridePendingTransition(R.anim.slide_in_up, R.anim.stay)
+            }
+
+        })
+        adapter!!.setOnLearnMoreButtonListener(object : WalletRecyclerViewAdapter.OnLearnMoreButtonListener {
+            override fun onLearnMoreButtonClicked(view: View, position: Int) {
+
+            }
+
+        })
+        walletRecyclerView.adapter = adapter
+        walletRecyclerView.layoutManager = LinearLayoutManager(activity)
+    }
+
+    private fun loadBalance() : String {
         val balances = WalletApplication.localStore!!.balances
 
         if (balances != null) {
             balances.forEach {
                 if (it.assetType == LUMENS_ASSET_TYPE) {
-                    balanceTextView.text = it.balance
+                    return it.balance
                 }
             }
+            return DEFAULT_ACCOUNT_BALANCE
         } else {
-            balanceTextView.text = DEFAULT_ACCOUNT_BALANCE
+            return DEFAULT_ACCOUNT_BALANCE
+        }
+    }
+
+    private fun displayNoNetwork() {
+        Toast.makeText(activity, getString(R.string.no_network), Toast.LENGTH_SHORT).show()
+
+    }
+
+    //endregion
+
+    //region Call backs
+
+    override fun onLoadAccount(result: AccountResponse?) {
+        if (result != null) {
+
+            result.balances?.forEach {
+                if (it.assetType == LUMENS_ASSET_TYPE) {
+                    recyclerViewArrayList!!.updateTotalBalance(TotalBalance(it.balance))
+                }
+            }
+
+            WalletApplication.localStore!!.balances = result.balances
+        }
+    }
+
+    override fun onLoadEffects(result: java.util.ArrayList<EffectResponse>?) {
+        if (result != null) {
+            recyclerViewArrayList!!.updateEffectsList(result)
+            adapter!!.notifyDataSetChanged()
         }
     }
 
+    //endregion
+
 
     //region API Polling
 
     private fun startPollingAccount() {
         runnableCode = object : Runnable {
             override fun run() {
 
-                Horizon.Companion.LoadAccountTask(this@WalletFragment)
-                        .execute(WalletApplication.session!!.keyPair)
+                if (WalletApplication.session != null && isNetworkAvailable()) {
```

Comment:
Nice call on network check. However, `isNetworkAvailable` would be better off in the `Horizon` services. If not available, a callback can send an error message saying `Network not available` and we can handle in UI by showing a snack bar.